extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -49,6 +49,9 @@ export class AmpAd extends AMP.BaseElement {
 
   /** @override */
   upgradeCallback() {
+    const getNow = (this.win.performance && this.win.performance.now) ?
+        this.win.performance.now.bind(this.win.performance) : Date.now;
+    const startTime = getNow();
```

Comment:
To reply the other comment: I think what @dvoytenko suggested is to keep the startTime in the element instead of passing around:

```
upgradeCallback() {
  ...
  this.element.upgradeStartTime_ = getNow();
}

... in A4A constructor:
const delta = getNow() - this.element.upgradeStartTime_;
```